### ISSUE TITLE
tsconfig w/ extends unable to locate sources using gulp-typescript

### ISSUE THREAD

**Issue by BurtHarris** [ivogabe/gulp-typescript#532]
Building with gulp-typescript I'm getting both an error message saying  "No inputs were found in config file 'tsconfig.test.json'..." and then an exception.

**Expected behavior:**
Locate included source files like TSC does
```
> tsc -p ./tsconfig.test.json
>
```

**Actual behavior:**
```
[11:09:17] Starting 'tsc.test'...
No inputs were found in config file 'tsconfig.test.json'. Specified 'include' paths were '["test/src/**/*.spec.js"]' and 'exclu
de' paths were '["node_modules"]'.
[11:09:17] 'tsc.test' errored after 4.43 ms
[11:09:17] Error: Invalid glob argument:
    at Object.src (C:\code\core-decorators.js\node_modules\gulp-typescript\node_modules\vinyl-fs\lib\src\index.js:32:11)
    at Function.src (C:\code\core-decorators.js\node_modules\gulp-typescript\release\project.js:96:16)
    at Gulp.<anonymous> (C:\code\core-decorators.js\gulpfile.js:17:22)
    at module.exports (C:\code\core-decorators.js\node_modules\orchestrator\lib\runTask.js:34:7)
    at Gulp.Orchestrator._runTask (C:\code\core-decorators.js\node_modules\orchestrator\index.js:273:3)
    at Gulp.Orchestrator._runStep (C:\code\core-decorators.js\node_modules\orchestrator\index.js:214:10)
    at C:\code\core-decorators.js\node_modules\orchestrator\index.js:279:18
    at finish (C:\code\core-decorators.js\node_modules\orchestrator\lib\runTask.js:21:8)
    at module.exports (C:\code\core-decorators.js\node_modules\orchestrator\lib\runTask.js:60:3)
    at Gulp.Orchestrator._runTask (C:\code\core-decorators.js\node_modules\orchestrator\index.js:273:3)
```
**Your gulpfile:**

*Include your gulpfile, or only the related task (with `ts.createProject`).*
```javascript
const gulp = require('gulp');
const runSequence = require('run-sequence');
const babel = require('gulp-babel');
const ts = require('gulp-typescript');
const sourcemaps = require('gulp-sourcemaps');
const del = require('del');
const exec = require('child_process').exec;

// Default build uses settings from 'tsconfig.json'

const tsProject = ts.createProject('tsconfig.json');
const tsProjectEsm = ts.createProject('tsconfig.esm.json');
const tsProjectTest = ts.createProject('tsconfig.test.json');
 
function tscTask( tsProject ) {
  return function() {
    return tsProject.src()
      .pipe(sourcemaps.init())
      .pipe(tsProject())
      .pipe(sourcemaps.write('.'))
      .pipe(gulp.dest(tsProject.config.compilerOptions.outDir))
  }
}

gulp.task('clean', function () {
  return del(['lib', 'esm', 'test/node_modules', 'test/typescript', 'test/babel']);
})

gulp.task('test_setup', function(cb) {
  return exec( "npm install", {cwd: "test"}, function(err, stdout, stderr) {
    console.log(stdout);
    console.log(stderr);
    cb(err);
  })
})

gulp.task('lib', tscTask(tsProject));
gulp.task('esm', tscTask(tsProjectEsm));
gulp.task('tsc.test', ['lib', 'test_setup'], tscTask(tsProjectTest));

gulp.task('src', ['lib', 'esm']);

gulp.task('babel.test', ['src', 'test_setup'], function () {
  return gulp.src('test/**/*.js')
    .pipe(sourcemaps.init())
    .pipe(babel({
      "presets": [["es2015", { "modules": false }]],
      "plugins": [
        "transform-object-rest-spread",
        "transform-flow-strip-types",
        "transform-decorators-legacy",
        "transform-class-properties"
      ],
    "env": {
      "es": {},
      "development": {
        "plugins": [
          "transform-es2015-modules-commonjs"
        ]
      }}}))
    .pipe(sourcemaps.write('.'))
    .pipe(gulp.dest('test/babel'))
})

gulp.task('test', ['babel.test', 'tsc.test'] )

gulp.task('watch', ['src','test'], function () {
  gulp.watch('src/**/*', ['src', 'test'])
  gulp.watch('test/**/*', ['test'])
});

gulp.task('default', function(cb) { 
  runSequence('clean', ['src', 'test'], cb);
})
```

**tsconfig.test.json**

```json
{
    "extends": "./tsconfig.json",
    "include": ["test/src/**/*.spec.js"],
    "exclude": ["node_modules"],
    "compilerOptions": {
        "outDir": "test/typescript"
    }
}
```

**tsconfig.json**

```json
{
  "include": ["src/**/*.ts", "src/**/*.js"],
  "compilerOptions": {
    "outDir": "lib",
    "target": "es5",
    "module":"commonjs",
    "sourceMap": true,
    "allowJs": true,
    "checkJs": true,
    "allowUnreachableCode": true,
    "lib": [
      "es5",
      "dom",
      "es2015","es2015.core", "es2015.collection"
    ],
    "baseUrl": ".",
    "paths": {
      "lib/*": ["src/*"]
    },
    "rootDirs": [
      "src", "test", "lib"
    ],
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true
  }
}
```

**Code**

*Include your TypeScript code, if necessary.*
```typescript

```

**Comment by ivogabe** [comment]
Support for 'extends' was added in version 3.2.0 and some bugs of it were fixed in 3.2.1. Could you check whether you're using that version?

**Comment by BurtHarris** [comment]
Thanks @ivogabe.   I dropped the branch I had using gulp-typescript but I'm getting ready to restart in that direction again.   I trust you've corrected the issue so I'll close for now.
